### ISSUE TITLE
max ded prior level allocation in fmcalc out of sync with fmpy

### ISSUE THREAD

**Issue by johcarter** [OasisLMF/ktools#235]
## Issue Description

The fmcalc alloc rule 2 (prior level) is out of sync with fmpy with respect to policies with maximum deductibles. When a maximum deductible is enforced and the loss is adjusted upwards, in fmcalc it can sometimes be back-allocated to zero (leading to understated losses), or back-allocated to exceed prior level limits or even input losses.

fmpy has an improved back-allocation methodology for a2 now, which back-allocates the delta (increase in loss due to a max ded) on the basis of the underlimit. The underlimit is defined as the amount of loss that any group of items can be increased by in the event of an upwards loss adjusted, taking into account input loss and any prior level limits.  This ensures that back-allocated losses never exceed prior level limits, input loss, and are never zero.

The pre-adjustment loss is back-allocated based on prior level in the same way as fmcalc.

If 
Loss = Pre-adjusted loss + Delta loss

then

fmcalc method (a2)
Back-allocated loss = Prior level weighting * Loss

fmpy method (a2)
Back-allocated loss = Prior level weighting * Pre-adjusted loss + Underlimit weighting * Delta loss

When Delta loss = 0, the methods are the same.

https://github.com/OasisLMF/OasisLMF/issues/803
https://github.com/OasisLMF/OasisLMF/pull/898

## Steps to Reproduce (Bugs only)

[Worked example policy calculation_Max deductibles.xlsx](https://github.com/OasisLMF/ktools/files/7127883/Worked.example.policy.calculation_Max.deductibles.xlsx)

**Comment by johcarter** [comment]
PR #236 aligns the methodology of adjusting loss for min and max deductibles to be in line with fmpy, but the back-allocation method is still out of sync.

**Comment by johcarter** [comment]
fixed in fmpy
